superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -125,8 +125,16 @@ class VisualizeModal extends React.PureComponent {
     this.setState({ columns }, this.validate);
   }
   render() {
-    if (!(this.props.query)) {
-      return <div />;
+    if (!(this.props.query) || !(this.props.query.results)) {
+      return (
+        <div className="VisualizeModal">
+          <Modal show={this.props.show} onHide={this.props.onHide}>
+            <Modal.Body>
+              No results available for this query
```

Comment:
it would be nice to have actionable feedback for the user here, but if there are no results for a query, re-running it won't likely return new results. i think we could save figuring out what actionable feedback we could show here for another PR.